Mathlib/Analysis/SpecialFunctions/Integrals/PosLogEqCircleAverage.lean

```diff
@@ -120,3 +154,110 @@ theorem circleAverage_log_norm_sub_const₁ (h : ‖a‖ = 1) :
+-/
+@[simp]
+theorem circleAverage_log_norm_sub_const_eq_posLog :
+    circleAverage (log ‖· - a‖) 0 1 = log⁺ ‖a‖ := by
+  rcases lt_trichotomy 1 ‖a‖ with h | h | h
+  · rw [circleAverage_log_norm_sub_const₂ h]
+    apply (posLog_eq_log _).symm
+    simp_all [le_of_lt h]
+  · rw [eq_comm, circleAverage_log_norm_sub_const₁ h.symm, posLog_eq_zero_iff]
+    simp_all
+  · rw [eq_comm, circleAverage_log_norm_sub_const₀ h, posLog_eq_zero_iff]
+    simp_all [le_of_lt h]
+
+/--
+The `circleAverage (log ‖· + a‖) 0 1` equals `log⁺ ‖a‖`.
+-/
+@[simp]
+theorem circleAverage_log_norm_add_const_eq_posLog :
+    circleAverage (log ‖· + a‖) 0 1 = log⁺ ‖a‖ := by
+  have : (log ‖· + a‖) = (log ‖· - -a‖) := by simp
+  simp [this]
+
+/--
+Generalization of `circleAverage_log_norm_sub_const_eq_posLog`: The
+`circleAverage (log ‖· - a‖) c R` equals `log R + log⁺ (|R|⁻¹ * ‖c - a‖)`.
+-/
+theorem circleAverage_log_norm_sub_const_eq_log_radius_add_posLog (hR : R ≠ 0) :
+    circleAverage (log ‖· - a‖) c R = log R + log⁺ (R⁻¹ * ‖c - a‖) := by
+  calc circleAverage (log ‖· - a‖) c R
+  _ = circleAverage (fun z ↦ log ‖R * (z + R⁻¹ * (c - a))‖) 0 1 := by
+    rw [circleAverage_eq_circleAverage_zero_one]
+    congr
+    ext z
+    congr
+    rw [Complex.ofReal_inv R]
+    field_simp [Complex.ofReal_ne_zero.mpr hR]
+    ring
+  _ = circleAverage (fun z ↦ log ‖R‖ + log ‖z + R⁻¹ * (c - a)‖) 0 1 := by
+    apply circleAverage_congr_codiscreteWithin _ (zero_ne_one' ℝ).symm
+    have : {z | ‖z + ↑R⁻¹ * (c - a)‖ ≠ 0} ∈ codiscreteWithin (Metric.sphere (0 : ℂ) |1|) := by
+      apply codiscreteWithin_iff_locallyFiniteComplementWithin.2
+      intro z hz
+      use Set.univ
+      simp only [univ_mem, abs_one, Complex.ofReal_inv, ne_eq, norm_eq_zero, Set.univ_inter,
+        true_and]
+      apply Set.Subsingleton.finite
+      intro z₁ hz₁ z₂ hz₂
+      simp_all only [ne_eq, abs_one, mem_sphere_iff_norm, sub_zero, Set.mem_diff, Set.mem_setOf_eq,
+        Decidable.not_not]
+      rw [add_eq_zero_iff_eq_neg.1 hz₁.2, add_eq_zero_iff_eq_neg.1 hz₂.2]
+    filter_upwards [this] with z hz
+    rw [norm_mul, log_mul (norm_ne_zero_iff.2 (Complex.ofReal_ne_zero.mpr hR)) hz]
+    simp
+  _ = log R + log⁺ (|R|⁻¹ * ‖c - a‖) := by
+    have : (fun z ↦ log ‖R‖ + log ‖z + ↑R⁻¹ * (c - a)‖) =
+        (fun z ↦ log ‖R‖) + (fun z ↦ log ‖z + ↑R⁻¹ * (c - a)‖) := by
+      rfl
+    rw [this, circleAverage_add (circleIntegrable_const (log ‖R‖) 0 1)
+      (circleIntegrable_log_norm_meromorphicOn (fun _ _ ↦ by fun_prop)), circleAverage_const]
+    simp
+  _ = log R + log⁺ (R⁻¹ * ‖c - a‖) := by
+    congr 1
+    rcases lt_trichotomy 0 R with h | h | h
+    · rw [abs_of_pos h]
+    · tauto
+    · simp [abs_of_neg h]
+
+/--
+Trivial corollary of
+`circleAverage_log_norm_sub_const_eq_log_radius_add_posLog`: If `u : ℂ` lies within the closed ball
+with center `c` and radius `R`, then the circle average
+`circleAverage (log ‖· - u‖) c R` equals `log R`.
+-/
+@[simp]
```

Comment:
I wouldn't make this one `@[simp]`, as there is a side condition which will typically be non-checkable by `simp`. Also, the name looks a little bit strange as there is no `logAbs` and no `affine` in the statement. 
Maybe `circleAverage_log_norm_sub_const_of_mem_closedBall`?

Reply:
Implemented as suggested. Thanks.